File: backend/app/schemas/tutor.py
from datetime import datetime
from pydantic import BaseModel
from typing import List, Literal, Optional

class ChatMessageBase(BaseModel):
    role: Literal["user", "assistant"]
    content: str

class ChatMessageRead(ChatMessageBase):
    id: int
    created_at: datetime

    class Config:
        from_attributes = True

class ChatSessionRead(BaseModel):
    id: int
    title: str
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class ChatRequest(BaseModel):
    session_id: Optional[int] = None
    messages: List[ChatMessageBase]

class ChatResponse(BaseModel):
    reply: str
    session_id: int

File: frontend/src/lib/api/tutor.ts
import { api } from './api';

export interface ChatMessage {
  id?: number;
  role: 'user' | 'assistant';
  content: string;
  created_at?: string;
}

export interface ChatSession {
  id: number;
  title: string;
  created_at: string;
  updated_at: string;
}

export interface ChatResponse {
  reply: string;
  session_id: number;
}

/**
 * Get all chat sessions for the current user.
 */
export async function getChatSessions(): Promise<ChatSession[]> {
  const response = await api.get<ChatSession[]>('/api/tutor/sessions');
  return response.data;
}

/**
 * Get all messages for a specific chat session.
 */
export async function getChatMessages(sessionId: number): Promise<ChatMessage[]> {
  const response = await api.get<ChatMessage[]>(`/api/tutor/sessions/${sessionId}/messages`);
  return response.data;
}

/**
 * Delete a specific chat session.
 */
export async function deleteChatSession(sessionId: number): Promise<void> {
  await api.delete(`/api/tutor/sessions/${sessionId}`);
}

/**
 * Send a message to the AI Tutor.
 * If sessionId is null, a new session is created on the backend.
 */
export async function sendChatMessage(messages: ChatMessage[], sessionId?: number | null): Promise<ChatResponse> {
  const payload = {
    session_id: sessionId || null,
    messages: messages.map(m => ({ role: m.role, content: m.content }))
  };
  const response = await api.post<ChatResponse>('/api/tutor/chat', payload);
  return response.data;
}
